File: eKucniLjubimci.Services/Database/Kupac.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace eKucniLjubimci.Services.Database
{
    public class Kupac
    {
        public int KupacId { get; set; }        
        public int BrojNarudzbi { get; set; }
        public bool Kuca { get; set; }
        public bool Dvoriste { get; set; }
        public bool Stan { get; set; }
        public bool isDeleted { get; set; } = false;


        public int OsobaId { get; set; }
        public virtual Osoba Osoba { get; set; }

        public virtual Lokacija Lokacija { get; set; }
        public int LokacijaId { get; set; }

        public virtual List<Narudzba> Narudzbe { get; set; }

        public string SlikaKupca { get; set; }

        public virtual KorisnickiNalog KorisnickiNalog { get; set; }
        public int KorisnickiNalogId { get; set; }
    }
}

File: flutterUI/flutter_mobile/lib/models/kupac.dart
import 'package:json_annotation/json_annotation.dart';

import 'korisnicki_nalog.dart';
import 'lokacija.dart';
import 'osoba.dart';

part 'kupac.g.dart';

@JsonSerializable()
class Kupac {
  int? kupacId;
  int? brojNarudzbi;
  bool? kuca;
  bool? dvoriste;
  bool? stan;
  Osoba? osoba;
  int? osobaId;
  Lokacija? lokacija;
  int? lokacijaId;
  List<dynamic>? narudzbe;
  String? slikaKupca;
  KorisnickiNalog? korisnickiNalog;
  int? korisnickiNalogId;

  Kupac({
    this.kupacId,
    this.brojNarudzbi,
    this.kuca,
    this.dvoriste,
    this.stan,
    this.osoba,
    this.osobaId,
    this.lokacija,
    this.lokacijaId,
    this.narudzbe,
    this.slikaKupca,
    this.korisnickiNalog,
    this.korisnickiNalogId,
  });
  factory Kupac.fromJson(Map<String, dynamic> json) => _$KupacFromJson(json);

  Map<String, dynamic> toJson() => _$KupacToJson(this);
}
